### PR TITLE
Enable APIs to use email typed date time strings

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1117

## Goal
To support email date-time typed string conversions.

## Approach
Enable APIs for the following use cases:
- convert a given UTC to an email string
- convert a given Civil to an email string
- convert a given email string to Civil

## Release note
Improve the time APIs to support email typed date-time string conversions.

## Automation tests
 - Unit tests 
   > Yes

## Related PRs
https://github.com/ballerina-platform/module-ballerina-time/pull/68

## Test environment
- macOS
- Java 11
- SL Alpha 3

## Learning
- https://stackoverflow.com/questions/2458524/parsing-rfc-2822-date-in-java
- https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

### CODE REVIEW

File: time-ballerina/time_types.bal [hunk 1]
```diff
@@ -39,36 +39,86 @@ public type Seconds decimal;
...
     int minute;
     Seconds second?;
 };
 
+# Date in proleptic Gregorian calendar with all the fields beign optional.
+#
+# + month - month as an integer(1 <= month <= 12)  
+# + year - year as an integer  
+# + day - day as an integer(1 <= day <= 31)  
+public type OptionalDateFields record {
```
Comment (daneshk) [inline time-ballerina/time_types.bal:86]: Do we need to make it public?

File: time-ballerina/time_types.bal [hunk 2]
```diff
@@ -39,36 +39,86 @@ public type Seconds decimal;
...
+    int month?;
+    int day?;
+};
+
+# TimeOfDay with all the fields beign optional.
+#
+# + hour - hour as an integer(0 <= hour <= 23)  
+# + minute - minute as an integer(0 <= minute <= 59)   
+# + second - second as decimal value with nanoseconds precision
+public type OptionalTimeOfDayFields record {
```
Comment (daneshk) [inline time-ballerina/time_types.bal:97]: Do we need to make it public?

**Review by daneshk — APPROVED** [review]

File: time-ballerina/time_types.bal (hunk 1 above)
Comment (BuddhiWathsala) [reply time-ballerina/time_types.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-time/pull/76/commits/466c2eab1f6ff3956bd25c0f278811c73fc8397a

File: time-ballerina/time_types.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply time-ballerina/time_types.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-time/pull/76/commits/466c2eab1f6ff3956bd25c0f278811c73fc8397a

**Pull request merged**